src/main/java/com/alex/eshop/service/UserService.java

```diff
@@ -13,28 +14,26 @@
 public class UserService {
 
     private final KeycloakService keycloakService;
-    private final KeycloakClientFactory keycloakClientFactory;
-    private final ApplicationProperties applicationProperties;
+    private final UserMapper userMapper;
 
-    public UserService(KeycloakService keycloakService, KeycloakClientFactory keycloakClientFactory, ApplicationProperties applicationProperties) {
+    public UserService(KeycloakService keycloakService, UserMapper userMapper) {
         this.keycloakService = keycloakService;
-        this.keycloakClientFactory = keycloakClientFactory;
-        this.applicationProperties = applicationProperties;
+        this.userMapper = userMapper;
     }
 
     public String getAccessToken() {
         AccessTokenResponse accessTokenResponse = keycloakService.getToken();
+
         return accessTokenResponse.getToken();
     }
 
-    public UserRepresentation getUserRepresentation(String userUuid) {
-        UserRepresentation userRepresentation = keycloakClientFactory.getInstance()
-                .realm(applicationProperties.getKeycloak().getRealm())
-                .users()
-                .get(userUuid)
-                .toRepresentation();
+    public UserDTO getUser(String userUuid) {
+        UserRepresentation userRepresentation = keycloakService.getUserRepresentation(userUuid);
 
-        return userRepresentation;
+        return userMapper.toDto(userRepresentation);
```

Comment:
use function `List<String> getUserRoles(String userUuid)` from `KeycloakService` and set roles here for `userDto`